modules/cognito_userpool/main.tf

```diff
@@ -0,0 +1,40 @@
+resource "aws_cognito_user_pool" "pool" {
+  name = "DataSimulationPool"
+  password_policy {
+    minimum_length    = 8
+    require_lowercase = true
+    require_uppercase = false  
+    require_symbols   = false
+    require_numbers   = true
+  }
+
+  device_configuration {
+    challenge_required_on_new_device = false
+  }
+
+  username_configuration {
+    case_sensitive = false
+  }
+
+  admin_create_user_config {
+    allow_admin_create_user_only = true
+  }
+}
+
+resource "aws_cognito_user_pool_client" "client" {
+  name            = "SimulationApp"
+  user_pool_id    = aws_cognito_user_pool.pool.id
+  generate_secret = false
```

Comment:
missing oauth scopes

Reply:
and auth flows

Reply:
I think this could be the fix

```tf
  allowed_oauth_flows = ["code", "implicit"]
  allowed_oauth_flows_user_pool_client = true
  allowed_oauth_scopes = ["email", "openid", "profile"]

  callback_urls = ["http://localhost"]
  default_redirect_uri = "http://localhost"

  explicit_auth_flows = [
    "ALLOW_USER_PASSWORD_AUTH",
    "ALLOW_REFRESH_TOKEN_AUTH"
  ]

  supported_identity_providers = ["COGNITO"]

  prevent_user_existence_errors = "ENABLED"

```